ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -1013,8 +1013,7 @@ func (r *RayClusterReconciler) createWorkerPod(ctx context.Context, instance ray
 // Build head instance pod(s).
 func (r *RayClusterReconciler) buildHeadPod(ctx context.Context, instance rayv1.RayCluster) corev1.Pod {
 	logger := ctrl.LoggerFrom(ctx)
-	podName := strings.ToLower(instance.Name + utils.DashSymbol + string(rayv1.HeadNode) + utils.DashSymbol)
-	podName = utils.CheckName(podName)                                            // making sure the name is valid
```

Comment:
I tried to change CheckName, but realized this would break clusters on upgrade because we also use `CheckName` for getting the RayCluster service name and the service name would change if previously truncated. 

So instead, I introduced a new function `GeneratePodName` that is only used for generating new pod names. I don't think this is a breaking change as old pods will continue to run and this is only run for newly created pods, but I will do some extra validation for this